src/game/creature.c

```diff
@@ -640,10 +644,6 @@ bool Creature_Animate(int16_t item_num, int16_t angle, int16_t tilt)
         }
 
         item->pos.x_rot = 0;
-
-        floor = Room_GetFloor(item->pos.x, item->pos.y, item->pos.z, &room_num);
-        item->floor =
-            Room_GetHeight(floor, item->pos.x, item->pos.y, item->pos.z);
```

Comment:
Can you explain this change, and why the LOT->fly `if` didn't need to be amended as well?

Reply:
This was changed in TR3+'s `CreatureVault`. Two things:
1. I moved everything to a `Creature_Vault` then ended up changing back. Because, we had to make changes compared to later TR's vaulting because the ape has no vault down animation. So `Ape_Vault` no longer matches. Also, Core has the `APE_TURN_L_FLAG` and `APE_TURN_R_FLAG` stuff in `Ape_Vault`. Now this could be extracted out in the future, but for now it matches the TR1 code more. (Btw, this flags are bugged I'm pretty sure but that's for another day/PR).

2. Basically, the ape has one chance to perform the vault anim which is when the ape crosses an edge (square in the game world). The original code which changed the `item->floor` after the `item->pos.y` updates was one frame behind, so the ape missed the window of opportunity to vault. They corrected this in future TR games where vaulting enemies worked properly. My guess is they ran out of time for TR1 to fix this or it was just an oversight because the ape technically went up/down blocks even though the vault animation didn't work.